typify-impl/src/lib.rs

```diff
@@ -493,7 +523,11 @@ impl TypeSpace {
         // Finalize all created types.
         for index in base_id..self.next_id {
             let type_id = TypeId(index);
-            let mut type_entry = self.id_to_entry.get(&type_id).unwrap().clone();
+            let mut type_entry = self
+                .id_to_entry
+                .get(&type_id)
+                .ok_or_else(|| Error::NoTypeEntryForId(type_id.clone()))?
```

Comment:
I think it's more helpful if this case panics; it reflects an actual logic but in this code.